signal/signals.go

```diff
@@ -44,9 +37,9 @@ func send(typ string, event interface{}) {
 		return
 	}
 
-	str := C.CString(string(data))
-	C.StatusServiceSignalEvent(str)
-	C.free(unsafe.Pointer(str))
+	notificationHandlerMutex.Lock()
```

Comment:
`RLock()` if we have `sync.RWMutex`. Regular mutex should also be fine.

Reply:
Right, good spot!